tbtadm/controller.cpp

```diff
@@ -653,6 +655,188 @@ catch (...)
     m_err << "Unknown exception\n";
 }
 
+void tbtadm::Controller::addToBootACL(const fs::path& dir, std::string uuid)
+{
+    auto fname = dir.filename().string();
+    auto num = fname.substr(0, fname.find('-'));
+    auto boot_acl_path = sysfsDevicesPath / (domain + num) / boot_acl;
+
+    if (!fs::exists(boot_acl_path))
+    {
+        /* Cannot find boot_acl */
+        return;
+    }
+
+    /* Read boot_acl */
+    std::stringstream stream(readAndTrim(boot_acl_path));
+    /* Check maximum allowed boot_acl string size */
+    if (stream.str().size() > 36 * 16 + 1 * 15)
+    {
+        m_out << "boot_acl contains too much characters\n";
+        return;
+    }
+
+    std::queue<std::string> acl_queue;
+
+    while(stream.good())
+    {
+        std::string substring;
+
+        getline(stream, substring, ',');
+	/* this moves also uuid to the end of the queue */
+        if (substring.size() == 36 && substring != uuid)
+        {
+            acl_queue.push(substring);
+        }
+    }
+
+    if (acl_queue.size() > 16)
+    {
+        m_out << "Too much elements in the boot_acl list\n";
+        return;
+    }
+
+    /* Add UUID to boot_acl queue */
+    if (acl_queue.size() < 16)
+    {
+        acl_queue.push(uuid);
+    }
+    else
+    {
+        m_out << "Trim boot_acl list, remove oldest entry\n";
+        acl_queue.pop();
+        acl_queue.push(uuid);
+    }
+
+    std::string out;
+
+    for (int i = 0; i < 16; i++)
+    {
+        if (!acl_queue.empty())
+        {
+            out.append(acl_queue.front());
+            acl_queue.pop();
+        }
+
+        if (i < 15)
+        {
+            out.append(",");
+        }
+    }
+
+    std::ofstream boot_acl_file(boot_acl_path.string());
+    boot_acl_file << out;
+    boot_acl_file.close();
+}
+
+std::string tbtadm::Controller::getDomain(std::string uuid)
```

Comment:
Maybe getDomainOfDevice is a better name?

Reply:
If we need to work with all controllers then we do not need this function.